refact-agent/gui/src/features/Chat/Thread/selectors.ts

```diff
@@ -136,9 +136,14 @@ export const selectHasUncalledTools = createSelector(
 
     if (toolCalls.length === 0) return false;
 
-    const toolMessages = tailMessages
-      .filter(isToolMessage)
-      .map((toolMessage) => toolMessage.content.tool_call_id);
+    const toolMessages = tailMessages.map((msg) => {
+      if (isToolMessage(msg)) {
+        return msg.content.tool_call_id;
+      }
+      if (isDiffMessage(msg)) {
+        return msg.tool_call_id;
```

Comment:
I think more types can have tool call id,
most will have it at the top level. so `if ("tool_call_id" in msg) { ... }`
also remember to filter them so theres no `undefined` values in the array